packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx

```diff
@@ -87,6 +89,27 @@ function TabsComponent(props: Props): JSX.Element {
   }
 
   const {tabGroupChoices, setTabGroupChoices} = useTabGroupChoice();
+  // search params >
+  // local storage >
+  // specified defaultValue >
+  // first child with "default" attr >
+  // first tab item.
+  let defaultValue: string | null | undefined;
```

Comment:
I'm not even sure we should consider the querystring as a default value

I believe the default value should rather be static (otherwise it could create weird react hydration issues because of a mismatch between server/client)

The actual state value could be "updated" to reflect the querystring in a `useEffect` after mount. I think the `useState` initial value should not be affected by a dynamic param such as the query string.

It may not be a problem today but we'll likely get a warning after upgrading to React 18 if we don't do this.

Reply:
Refactored into `useEffect` as requested.